execution/executor-benchmark/src/block_partitioning.rs

```diff
@@ -19,11 +18,17 @@ pub(crate) struct BlockPartitioningStage {
 }
 
 impl BlockPartitioningStage {
-    pub fn new(num_shards: usize) -> Self {
+    pub fn new(num_shards: usize, partition_last_round: bool) -> Self {
```

Comment:
i think `partition_last_round` is a param very specific to V1/V2. It makes more sense to either pass in a pointer of `<dyn BlockPartitioner>`, or, inside the new() function, `build_from_envvar()`.

Reply:
How about this as a follow up as well?